iothub/service/src/HttpRegistryManager.cs

```diff
@@ -959,58 +959,86 @@ public override Task CancelJobAsync(string jobId)
-            };
+            return ExportDevicesAsync(
+                JobProperties.CreateForExportJob(
+                    exportBlobContainerUri,
+                    excludeKeys,
+                    outputBlobName),
+                ct);
+        }
 
-            return CreateJobAsync(jobProperties, ct);
+        public override Task<JobProperties> ExportDevicesAsync(JobProperties jobParameters, CancellationToken cancellationToken = default)
+        {
+            jobParameters.Type = JobType.ExportDevices;
+            return CreateJobAsync(jobParameters, cancellationToken);
         }
 
         public override Task<JobProperties> ImportDevicesAsync(string importBlobContainerUri, string outputBlobContainerUri)
         {
-            return ImportDevicesAsync(importBlobContainerUri, outputBlobContainerUri, CancellationToken.None);
+            return ImportDevicesAsync(
+                JobProperties.CreateForImportJob(
+                    importBlobContainerUri,
+                    outputBlobContainerUri));
         }
 
         public override Task<JobProperties> ImportDevicesAsync(string importBlobContainerUri, string outputBlobContainerUri, CancellationToken ct)
         {
-            return ImportDevicesAsync(importBlobContainerUri, outputBlobContainerUri, null, ct);
+            return ImportDevicesAsync(
+                JobProperties.CreateForImportJob(
+                    importBlobContainerUri,
+                    outputBlobContainerUri),
+                ct);
         }
 
         public override Task<JobProperties> ImportDevicesAsync(string importBlobContainerUri, string outputBlobContainerUri, string inputBlobName)
         {
-            return ImportDevicesAsync(importBlobContainerUri, outputBlobContainerUri, inputBlobName, CancellationToken.None);
+            return ImportDevicesAsync(
+                JobProperties.CreateForImportJob(
+                    importBlobContainerUri,
+                    outputBlobContainerUri,
+                    inputBlobName));
         }
 
         public override Task<JobProperties> ImportDevicesAsync(string importBlobContainerUri, string outputBlobContainerUri, string inputBlobName, CancellationToken ct)
         {
-            var jobProperties = new JobProperties()
-            {
-                Type = JobType.ImportDevices,
-                InputBlobContainerUri = importBlobContainerUri,
-                OutputBlobContainerUri = outputBlobContainerUri,
-                InputBlobName = inputBlobName,
-            };
+            return ImportDevicesAsync(
+                JobProperties.CreateForImportJob(
+                    importBlobContainerUri,
+                    outputBlobContainerUri,
+                    inputBlobName),
+                ct);
+        }
 
-            return CreateJobAsync(jobProperties, ct);
+        public override Task<JobProperties> ImportDevicesAsync(JobProperties jobParameters, CancellationToken cancellationToken = default)
```

Comment:
New API impl for import.

All previous methods now call this one. It enforces the job type.

Reply:
On out side, do we not need to validate number of parameters required for ImportDevices - InputBlobContainerUri and OutputBlobContainerUri at least?

Reply:
We can let the service validate and refuse.